aptos-move/aptos-intent/src/decompiler.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright © Aptos Foundation
```

Comment:
Love this approach. One question though: when do we decide to run this decompiler? When we see some special flag in the metadata? I'm just thinking more about the unhappy path where you have something that isn't really an intent script.

Reply:
Also playing the devil's advocate: how do you envision the long term maintainability of this mini decompiler if we continue to add more functionalities?

Reply:
This is a very good point. Since the decompiler will be served as part of the full node we need extra eyes on this logic to make sure it won't crash anything. 

For longer term maintainability I think we need to maintain the tests to make sure the round trip equality holds?